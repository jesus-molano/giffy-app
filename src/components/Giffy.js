import { useState } from 'react';
import InputCategory from './InputCategory';
import GifGrid from './GifGrid';

const Giffy = ({defaultCategories = []}) => {
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<h1>Giffy</h1>
			<InputCategory setCategories={setCategories} />
			<hr />
			<div className="container">
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</div>
		</>
	);
};

export default Giffy;
