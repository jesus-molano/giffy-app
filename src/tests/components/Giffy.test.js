import { shallow } from 'enzyme';
import Giffy from '../../components/Giffy';

describe('Principal component tests', () => {
  test('should show the component correctly', () => {
    const wrapper = shallow(<Giffy />)
    expect(wrapper).toMatchSnapshot();
  })
  test('should show a list of categories', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<Giffy defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  })
  
})

