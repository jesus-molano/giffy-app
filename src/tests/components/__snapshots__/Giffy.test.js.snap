// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Principal component tests should show a list of categories 1`] = `
<Fragment>
  <h1>
    Giffy
  </h1>
  <InputCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="container"
  >
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;

exports[`Principal component tests should show the component correctly 1`] = `
<Fragment>
  <h1>
    Giffy
  </h1>
  <InputCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="container"
  />
</Fragment>
`;
